target/targets.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import dataclass
+from typing import Optional
+
+import serial
+
+from target.cw_fpga import CWFPGA
+
+
+@dataclass
+class TargetConfig:
+    """ Target configuration.
+    Stores information about the target.
+    """
+    target_type: str
+    fw_bin: str
+    protocol: str
+    pll_frequency: int
+    output_len: int
+    bitstream: Optional[str] = None
+    force_program_bitstream: Optional[bool] = False
+    baudrate: Optional[int] = None
+    port: Optional[str] = None
+    read_timeout: Optional[int] = 1
+
+
+class Target:
+    """ Target class.
+
+    Represents a SCA/FI target. Currently, either a ChipWhisperer FPGA board
+    or the discrete chip are supported.
```

Comment:
Ulta nit: this PR doesn't yet add support for the discrete chip ;-)

Reply:
Thanks, updated.